[Wee Zhuo Lin] iP

jolwnn opened the pull request:
# Rudolf: Your Christmas Task Companion 🎄

> “Christmas isn't a season. It's a feeling.” – Edna Ferber

**Rudolf** is a Java-based chatbot designed to help you manage your Christmas season tasks with a festive twist. It supports different types of tasks, such as `ToDos`, `Deadlines`, and `Events`, allowing you to keep track of everything from gift shopping to holiday parties. Rudolf is here to make sure your Christmas is as organized and joyful as possible!

### Key Features

- **Manage Tasks**: Add, list, and delete tasks effortlessly.
- **Mark Tasks as Done**: Keep track of what you've accomplished.
- **Task Types**: Support for `ToDos`, `Deadlines`, and `Events`.
- **Christmas-Themed Responses**: Enjoy festive interactions as you manage your tasks.

### How to Use Rudolf

1. **Add Tasks**: Use commands like `add n/Wrap gifts p/12345678 e/wrap@tasks.com a/Living room t/Holiday` to add tasks.
2. **List Tasks**: Type `list` to see all your current tasks.
3. **Mark as Done**: Mark tasks as done using the `done` command followed by the task number.
4. **Find Tasks**: Search for specific tasks using keywords with the `find` command.

![image](https://github.com/user-attachments/assets/81308a1a-e129-4695-812a-bd04fa794c86)

If you are a Java programmer, you can use it to practice Java too. Here's the main method in the Duke class:
```java
public class Duke {
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
}
```

### To-Do List

- [x] Implement basic task management
- [x] Add Christmas-themed responses
- [x] Enhance UI with more Christmas spirit

Whether it's wrapping presents, sending out cards, or hosting a Christmas dinner, let Rudolf and the helpful elves keep track of everything so you can enjoy the magic of Christmas without a hitch. 🎅

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jolwnn committed:
Modify greet and Add exit

jolwnn committed:
Add echo

jolwnn committed:
Add feature to add, list tasks

jolwnn committed:
Add Mark as Done and Unmark feature

jolwnn committed:
Add ToDos, Events, Deadlines feature

jolwnn committed:
Add Error Handling

jolwnn committed:
Add Delete Tasks feature

jolwnn committed:
Modified input and expected for tests

jolwnn committed:
Add feature to save tasks

jolwnn committed:
Add feature to save tasks retry

jolwnn committed:
Merge branch 'branch-Level-7'

jolwnn committed:
Update to use LocalDateTime for dates

jolwnn committed:
Merge branch 'branch-Level-8'

jolwnn committed:
Make the code more OOP

jolwnn committed:
Organize into Packages

jolwnn committed:
Organize into packages retry

jolwnn committed:
Merge branch 'add-gradle-support'

jolwnn committed:
Add JUnit tests

jolwnn committed:
Add JavaDoc comments

jolwnn committed:
Make code follow coding standard

jolwnn committed:
Add find task by keyword function

jolwnn committed:
Merging branch-A-JavaDoc into master

jolwnn committed:
Merging branch-A-CodingStandard into master

jolwnn committed:
Fix merge conflict

lavanyagarg112 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
its great to see that you followed the boolean naming convention

lavanyagarg112 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+ * Inherits from the Task class.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
you can try making the variable name more descriptive perhaps?

lavanyagarg112 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,96 @@
+ * This class provides methods to manage tasks in a list, including adding, removing, and retrieving tasks.
+ */
+public class TaskList {
+    private final List<Task> tasks;
```
good to see that the plural "tasks" has been used as per the coding standard for naming

lavanyagarg112 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,213 @@
+     *
+     * @param input The user's input as a string.
+     */
+    public void parse(String input) {
+        if (input.equals("list")) {
+            ui.showTaskList(tasks.getTasks());
+        } else if (input.startsWith("mark")) {
+            markTask(input);
+        } else if (input.startsWith("unmark")) {
+            unmarkTask(input);
+        } else if (input.startsWith("todo")) {
+            addTodo(input);
+        } else if (input.startsWith("deadline")) {
+            addDeadline(input);
+        } else if (input.startsWith("event")) {
+            addEvent(input);
+        } else if (input.startsWith("delete")) {
+            deleteTask(input);
+        } else if (input.startsWith("find")) {
+            handleFind(input);
+        } else if (input.equals("bye")) {
+            ui.showGoodbyeMessage();
+        } else {
+            ui.showUnknownCommand();
+        }
+    }
+
```
these are descriptive method names, looks good!

lavanyagarg112 reviewed (commented):
The code quality looks good to me

jolwnn committed:
Add checkstyle

jolwnn committed:
Merge branch 'branch-A-CheckStyle'

jolwnn committed:
Add GUI

jolwnn committed:
Merge branch 'branch-Level-10'

jolwnn committed:
Improve code quality by following Java standards

The codebase is missing Javadoc comments  for some public classes and methods., includes unused imports, and has methods not named according to Java conventions.

This needs to change to enhance code readability and maintainability.

Add Javadoc comments, remove unused imports, and rename methods to use verb-based names.

This ensures consistency and clarity throughout the codebase.

jolwnn committed:
Add assertions for better code validation

The code lacks
assertions to ensure critical data and object states are valid.

Add assertions to verify non-null and non-empty values for file paths,
greeting messages, user inputs, and responses.

This enhances code robustness by catching issues early and ensuring
correct behavior during development.

jolwnn committed:
Merge pull request #2 from jolwnn/branch-A-CodeQuality

Improve code quality

jolwnn committed:
Merge branch 'master' into branch-A-Assertions

jolwnn committed:
Merge pull request #3 from jolwnn/branch-A-Assertions

Add assertions for better code validation

jolwnn committed:
Add feature to list tasks sorted by date or description

jolwnn committed:
Rename mentions of duke to rudolf

jolwnn committed:
Add Ui screenshot

jolwnn committed:
Update README.md

jolwnn committed:
Update README.md

jolwnn committed:
Update README.md

jolwnn committed:
Add user guide

jolwnn committed:
Merge branch 'master' of https://github.com/jolwnn/ip

jolwnn committed:
Use filepath for ui image

jolwnn committed:
Update ui image and user guide

jolwnn committed:
Update build.gradle

jolwnn committed:
Update user guide

jolwnn committed:
Update user guide

jolwnn committed:
Fix application not exiting on bye and update user guide

jolwnn committed:
Update README.md

jolwnn committed:
Update user guide

jolwnn committed:
Merge branch 'master' of https://github.com/jolwnn/ip
